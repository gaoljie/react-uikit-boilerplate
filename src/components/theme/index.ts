import React from "react";

export const theme = {
  primary: "#EF6F35",
  primaryFont: "Montserrat",
  secondaryFont: "Work Sans",
  desktop: "@media (max-width: 1920px)",
  smallDesktop: "@media (max-width: 1441px)",
  tablet: "@media (max-width: 769px)",
  mobile: "@media (max-width: 600px)"
};

export const ThemeContext = React.createContext(theme);

export const ThemeProvider = ThemeContext.Provider;
