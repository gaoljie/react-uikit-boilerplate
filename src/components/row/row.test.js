import React from "react";
import "jest-styled-components";
import { mount } from "enzyme";
import theme from "../theme";
import Row from "./index";

const props = {
  as: "p",
  dp: "flex",
  fd: "column",
  bs: "border-box",
  width: "180px",
  height: "100px",
  fs: "18px",
  fw: "wrap",
  jc: "flex-start",
  ai: "flex-end",
  ji: "start",
  bg: "white",
  m: "10px",
  ta: "center",
  lh: "1",
  span: 4,
  lg: 3,
  md: 2,
  sm: 1,
  start: 2,
  color: "black",
  p: "10px",
  ml: "20px",
  mr: "21px",
  mt: "22px",
  mb: "23px",
  pl: "24px",
  pr: "25px",
  pt: "26px",
  pb: "27px",
  children: <span>children</span>
};
describe("Test Row", () => {
  it("Matches the snapshot", () => {
    const wrapper = mount(
      <Row>
        <div>content</div>
      </Row>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("Styling", () => {
    const wrapper = mount(<Row {...props} />);

    expect(wrapper).toHaveStyleRule("grid-template-columns", "repeat(12,1fr)");

    expect(wrapper).toHaveStyleRule("grid-template-columns", `repeat(6,1fr)`, {
      media: theme.tablet.slice(7)
    });
    expect(wrapper).toHaveStyleRule("grid-template-columns", `repeat(2,1fr)`, {
      media: theme.mobile.slice(7)
    });
  });
});
