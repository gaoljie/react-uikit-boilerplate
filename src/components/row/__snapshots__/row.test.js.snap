// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Row Matches the snapshot 1`] = `
.c0 {
  display: grid;
  grid-template-columns: repeat(12,1fr);
  grid-column-gap: 20px;
  padding-left: calc((100vw - 1920px) / 2 + 120px);
  padding-right: calc((100vw - 1920px) / 2 + 120px);
}

@media (max-width:1920px) {
  .c0 {
    padding-left: 120px;
    padding-right: 120px;
  }
}

@media (max-width:1441px) {
  .c0 {
    padding-left: 80px;
    padding-right: 80px;
  }
}

@media (max-width:769px) {
  .c0 {
    padding-left: 40px;
    padding-right: 40px;
    grid-template-columns: repeat(6,1fr);
  }
}

@media (max-width:600px) {
  .c0 {
    padding-left: 20px;
    padding-right: 20px;
    grid-template-columns: repeat(2,1fr);
  }
}

<Styled(styled.div)>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  ",
            [Function],
            "
",
            "
  display: grid;
  grid-template-columns: repeat(12, 1fr);
  grid-column-gap: 20px;
  padding-left: calc((100vw - 1920px) / 2 + 120px);
  padding-right: calc((100vw - 1920px) / 2 + 120px);

  ",
            "@media (max-width: 1920px)",
            " {
    padding-left: 120px;
    padding-right: 120px;
  }
  ",
            "@media (max-width: 1441px)",
            " {
    padding-left: 80px;
    padding-right: 80px;
  }

  ",
            "@media (max-width: 769px)",
            " {
    padding-left: 40px;
    padding-right: 40px;
    grid-template-columns: repeat(6, 1fr);
  }

  ",
            "@media (max-width: 600px)",
            " {
    padding-left: 20px;
    padding-right: 20px;
    grid-template-columns: repeat(2, 1fr);
  }
",
          ],
        },
        "displayName": "Styled(styled.div)",
        "foldedComponentIds": Array [
          "sc-bdVaJa",
        ],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="c0"
    >
      <div>
        content
      </div>
    </div>
  </StyledComponent>
</Styled(styled.div)>
`;
